src/engine/gameObject/gameObject.js

```diff
@@ -1,6 +1,16 @@
 const COMPONENT_ADDED = 'COMPONENT_ADDED';
 const COMPONENT_REMOVED = 'COMPONENT_REMOVED';
 
+const findAncestor = (gameObject) => {
```

Comment:
Here instead of creating the function every time on `getAncestor()`, we only point to the created function which should work faster